src/test/app/Offer_test.cpp

```diff
@@ -795,22 +850,831 @@ class Offer_test : public beast::unit_test::suite
+
+        //WS client is used here because the RPC client could not
+        //be convinced to pass the build_path argument
+        auto wsc = makeWSClient(env.app().config());
+        Json::Value payment;
+        payment[jss::secret] = toBase58(generateSeed("bob"));
+        payment[jss::id] = env.seq (bob);
+        payment[jss::build_path] = true;
+        payment[jss::tx_json] = pay (bob, bob, bob["XXX"] (1));
+        payment[jss::tx_json][jss::Sequence] =
+            env.current ()->read (
+                keylet::account (bob.id ()))->getFieldU32 (sfSequence);
+        payment[jss::tx_json][jss::Fee] =
+            std::to_string( env.current ()->fees ().base);
+        payment[jss::tx_json][jss::SendMax] =
+            bob ["XTS"] (1.5).value ().getJson (0);
+        auto jrr = wsc->invoke("submit", payment);
+        BEAST_EXPECT(jrr[jss::status] == "success");
+        BEAST_EXPECT(jrr[jss::result][jss::engine_result] == "tesSUCCESS");
+
+        jrr = ledgerEntryState (env, alice, gw, "XTS");
+        BEAST_EXPECT(jrr[jss::node][sfBalance.fieldName][jss::value] == "-101");
+        jrr = ledgerEntryState (env, alice, gw, "XXX");
+        BEAST_EXPECT(jrr[jss::node][sfBalance.fieldName][jss::value] == "-99");
+
+        jrr = ledgerEntryState (env, bob, gw, "XTS");
+        BEAST_EXPECT(jrr[jss::node][sfBalance.fieldName][jss::value] == "-99");
+        jrr = ledgerEntryState (env, bob, gw, "XXX");
+        BEAST_EXPECT(jrr[jss::node][sfBalance.fieldName][jss::value] == "-101");
+    }
+
     void run ()
     {
-        testCanceledOffer ();
-        testRmFundedOffer ();
-        testTinyPayment ();
-        testXRPTinyPayment ();
-        testEnforceNoRipple ();
-        testInsufficientReserve ();
-        testFillModes ();
-        testMalformed ();
-        testExpiration ();
-        testUnfundedCross ();
+        testcaseCanceledOffer ();
+        testcaseRmFundedOffer ();
+        testcaseTinyPayment ();
+        testcaseXRPTinyPayment ();
+        testcaseEnforceNoRipple ();
+        testcaseInsufficientReserve ();
+        testcaseFillModes ();
+        testcaseMalformed ();
+        testcaseExpiration ();
+        testcaseUnfundedCross ();
+        testcaseSelfCross ();
+        testcaseNegativeBalance();
+        testcaseOfferCrossWithXRP(true);
+        testcaseOfferCrossWithXRP(false);
+        testcaseOfferCrossWithLimitOverride();
+        testcaseOfferAcceptThenCancel();
+        testcaseOfferCancelPastAndFuture();
+        testcaseCurrencyConversionEntire();
+        testcaseCurrencyConversionIntoDebt();
+        testcaseCurrencyConversionInParts();
+        testcaseCrossCurrencyStartXRP();
+        testcaseCrossCurrencyEndXRP();
+        testcaseCrossCurrencyBridged();
+        testcaseOfferFeesConsumeFunds();
+        testcaseOfferCreateThenCross();
+        testcaseSellFlagBasic();
+        testcaseSellFlagExceedLimit();
+        testcaseGatewayCrossCurrency();
```

Comment:
Nit: naming these `testcaseXXX` instead of `testXXX` only increases the length of the name without providing any clarity. I'd recommend renaming them back to `testXXX` for consistency with the rest of the code if nothing else.